libminifi/src/ThreadedSchedulingAgent.cpp

```diff
@@ -64,11 +68,18 @@ void ThreadedSchedulingAgent::schedule(Processor *processor)
 		return;
 	}
 
+	auto processContext = std::make_shared<ProcessContext>(processor);
```

Comment:
@phrocker Mainly, creating a shared ProcessContext and SessionFactory which will be used in this thread as well as the threads created in the loop below. I wanted these objects to be cleaned up once they go out of scope in all threads. The raw pointers are only used to integrate with the existing Processor pointer that is passed-in, as well as passing raw pointers to the constructors of the new shared objects. I wanted to minimize function/constructor changes that might have impacts elsewhere in the code base. If we want to move more to having managed pointers across the code base, then some of this can be cleaned up.